src/main/java/frc/team2412/robot/subsystems/ArmSubsystem.java

```diff
@@ -88,60 +127,78 @@ public static class ArmConstants { // To find values later
 	private final ProfiledPIDController armPID;
 	private final ProfiledPIDController wristPID;
 
+	private final ArmFeedforward armFeedforward;
+	private final ArmFeedforward wristFeedforward;
+
 	// Constructor
 
 	public ArmSubsystem() {
 		armMotor = new CANSparkMax(ARM_MOTOR, MotorType.kBrushless);
 		wristMotor = new CANSparkMax(WRIST_MOTOR, MotorType.kBrushless);
+
 		shoulderEncoder = new Encoder(SHOULDER_ENCODER_PORT_A, SHOULDER_ENCODER_PORT_B);
 		wristEncoder = new Encoder(WRIST_ENCODER_PORT_A, WRIST_ENCODER_PORT_B);
 
 		armPID = new ProfiledPIDController(ARM_K_P, ARM_K_I, ARM_K_D, ARM_CONSTRAINTS);
 		wristPID = new ProfiledPIDController(WRIST_K_P, WRIST_K_I, WRIST_K_D, WRIST_CONSTRAINTS);
 
+		armFeedforward = new ArmFeedforward(ARM_K_A, ARM_K_G, ARM_K_S, ARM_K_V);
+		wristFeedforward = new ArmFeedforward(WRIST_K_A, WRIST_K_G, WRIST_K_S, WRIST_K_V);
+
 		armMotor.setIdleMode(IdleMode.kBrake);
 		wristMotor.setIdleMode(IdleMode.kBrake);
 		armPID.setTolerance(ARM_POS_TOLERANCE, ARM_VELOCITY_TOLERANCE);
 		wristPID.setTolerance(WRIST_POS_TOLERANCE, WRIST_VELOCITY_TOLERANCE);
 
+		currentPosition = LOW;
+		manualOverride = false;
+
 		shoulderEncoder.reset();
 		wristEncoder.reset();
 	}
 
 	// Methods
 
-	// might not need?
-	// public double getVerticalArmPos() {
-	// 	return ((ARM_LENGTH * Math.sin(getShoulderAngle()))
-	// 			+ (VIRTUAL_BAR_LENGTH * Math.sin(getElbowAngle())));
-	// }
+	public void setArmMotor(double percentOutput) {
+		armMotor.set(MAX_ARM_SPEED * percentOutput);
+	}
 
-	// public double getHorizontalArmPos() {
-	// 	return ((ARM_LENGTH * Math.cos(getShoulderAngle()))
-	// 			+ (VIRTUAL_BAR_LENGTH * Math.cos(getElbowAngle())));
-	// }
+	public void setWristMotor(double percentOutput) {
+		wristMotor.set(MAX_WRIST_SPEED + percentOutput);
+	}
 
-	/** Stops Arm */
-	public void stopArm() {
-		armMotor.stopMotor();
+	public void setPosition(PositionType position) {
+		currentPosition = position;
 	}
 
-	public void stopWrist() {
-		wristMotor.stopMotor();
+	public void setManualOverride(boolean override) {
+		manualOverride = override;
 	}
 
-	public void rotateArmTo(double targetAngle) {
-		if (getShoulderAngle() != targetAngle) {
-			armPID.setGoal(targetAngle);
-			armMotor.set(armPID.calculate(getShoulderAngle(), targetAngle));
-		}
+	public void setArmGoal(double targetAngle) {
```

Comment:
We don't want to allow values outside of our range of motion to be set.  Is this the right place to add a check?